price-alerts-and-updates/update_price_and_send_alerts.py

```diff
@@ -0,0 +1,377 @@
+            SET product_availability = %s
+            WHERE product_id = %s
+            """
+
+CHECK_AVAILABILITY_QUERY = """
+            SELECT product_availability FROM products WHERE product_id = %s
+            """
+
+EMAIL_QUERY = """
+            SELECT users.email FROM users 
+            FULL OUTER JOIN subscriptions ON users.user_id = subscriptions.user_id 
+            WHERE subscriptions.product_id = %s
+            """
+
+INSERT_PRICE_QUERY = """
+            INSERT INTO prices (updated_at, product_id, price) VALUES (%s, %s, %s);
+            """
+
+GET_ALL_PRODUCTS_QUERY = "SELECT * FROM products;"
+
+GET_LATEST_PRICE_QUERY = """
+            SELECT price FROM prices WHERE product_id = (%s) 
+            ORDER BY updated_at DESC LIMIT 1;
+            """
+
+
+def get_database_connection() -> connection:
+    """
+    Return a connection of database.
+    """
+    try:
+        return connect(
+            user=environ["DB_USER"],
+            password=environ["DB_PASSWORD"],
+            host=environ["DB_HOST"],
+            port=environ["DB_PORT"],
+            database=environ["DB_NAME"]
+        )
+    except ConnectionError as error:
+        return error
+
+
+def create_ses_client() -> boto3.client:
+    """
+    Create and return a Boto3 client for AWS SES using AWS credentials.
+    """
+    ses_client = boto3.client(
+        'ses',
+        aws_access_key_id=environ["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=environ["AWS_SECRET_ACCESS_KEY"],
+        region_name="eu-west-2"
+    )
+    return ses_client
+
+
+def get_all_product_data(rds_conn: connection) -> extras.RealDictRow:
+    """
+    Query database for data on all products.
+    """
+    cur = rds_conn.cursor(cursor_factory=extras.RealDictCursor)
+    cur.execute(GET_ALL_PRODUCTS_QUERY)
+    rows = cur.fetchall()
+    cur.close()
+
+    new_rows = [dict(row) for row in rows]
+    return new_rows
+
+
+def get_user_data(rds_conn: connection, product_id: int) -> list:
+    """
+    Query database for users which are subscribed to given product.
+    """
+    cur = rds_conn.cursor(cursor_factory=extras.RealDictCursor)
+    cur.execute(EMAIL_QUERY, (product_id,))
+    rows = cur.fetchall()
+    cur.close()
+
+    return [entry['email'] for entry in rows]
+
+
+def update_product_availability(rds_conn: connection, product,
+                                availability: bool,
+                                ses_client: boto3.client) -> None:
+    """
+    Update product table to reflect availability of item as shown on webpage.
+    Emails users about a change in availability. 
+    """
+
+    with rds_conn.cursor(cursor_factory=extras.RealDictCursor) as cur:
+        cur.execute(UPDATE_AVAILABILITY_QUERY,
+                    (availability, product['product_id']))
+
+    rds_conn.commit()
+
+    recipients = get_user_data(rds_conn, product['product_id'])
+
+    if availability == True and len(recipients) >= 1:
+        for recipient in recipients:
```

Comment:
I think here you don't need a for loop, if you pass in `recipients` in the dictionary of `Destination` - I think it will send the same email to everyone in the list :)